File: New folder/cdp-sdk-8fcfac348487f586b224b77074f52594ad44198b/examples/python/solana/accounts/import_account.py
# Usage: uv run python solana/accounts/import_account.py
import asyncio
import base58
from cdp import CdpClient
from dotenv import load_dotenv
from solders.keypair import Keypair


async def main():
     async with CdpClient() as cdp:
        keypair = Keypair()
        
        # Import using base58 encoded private key
        print("1. Importing account using base58 encoded private key:")
        private_key_base58 = base58.b58encode(keypair.to_bytes()).decode('utf-8')
        print(f"Private key (base58): {private_key_base58}")
        
        account1 = await cdp.solana.import_account(
            private_key=private_key_base58,
        )
        print(f"Account imported successfully: {account1.address}\n")
        
        # Import using 32 byte private key array
        print("2. Importing account using 32 byte private key array:")
        new_keypair = Keypair()
        private_key_32_bytes = new_keypair.to_bytes()[:32]  # Take first 32 bytes (seed)
        print(f"Private key length: {len(private_key_32_bytes)} bytes")
        print(f"Private key (hex): {private_key_32_bytes.hex()}")
        
        account2 = await cdp.solana.import_account(
            private_key=private_key_32_bytes,
        )
        print(f"Account imported successfully: {account2.address}\n")
            
asyncio.run(main())

File: New folder/cdp-sdk-8fcfac348487f586b224b77074f52594ad44198b/examples/typescript/solana/accounts/importAccount.ts
// Usage: pnpm tsx solana/accounts/importAccount.ts

import { CdpClient } from "@coinbase/cdp-sdk";
import "dotenv/config";
import { Keypair } from "@solana/web3.js";
import bs58 from 'bs58';

const cdp = new CdpClient();

// Importing account with base58 encoded private key (64 bytes)
console.log("--------------------------------");
console.log("Importing account with 64-byte private key...");
const keypair = Keypair.generate();
const privateKey = bs58.encode(keypair.secretKey); // secretKey is 64 bytes (32 bytes private + 32 bytes public)

const account = await cdp.solana.importAccount({
  privateKey: privateKey, // e.g. "3MLZ...Uko8zz"
});

console.log("Imported account (64-byte key):", account.address);

// Verify the imported key length
const keyBytes64 = bs58.decode(privateKey);
console.log("Original private key length:", keyBytes64.length, "bytes");

// Importing account with 32-byte array private key
console.log("--------------------------------");
console.log("Importing account with raw bytes directly (32-byte)...");
const secondKeypair = Keypair.generate();
const privateKeyBytes32 = secondKeypair.secretKey.subarray(0, 32); // Take only first 32 bytes as Uint8Array

const secondAccount = await cdp.solana.importAccount({
  privateKey: privateKeyBytes32, // Using raw bytes directly instead of base58 string
  name: "BytesAccount32",
});

console.log("Imported account (raw 32-byte):", secondAccount.address);
console.log("Raw private key length:", privateKeyBytes32.length, "bytes");

console.log("--------------------------------");
console.log("All accounts imported successfully!");
console.log("64-byte string account address:", account.address);
console.log("32-byte bytes account address:", secondAccount.address);